dataset/batch_image.py

```diff
@@ -1,525 +1,767 @@
-    def get_image_size(self, image):
-        """ Return an image size (width, height) """
-        raise NotImplementedError("Should be implemented in child classes")
 
-    @action
-    def crop(self, components='images', origin=None, shape=None):
-        """ Crop all images in the batch
+class BaseImagesBatch(Batch):
+    """ Batch class for 2D images """
+    components = "images", "labels"
+    #pylint: disable=arguments-differ
+    def _assemble(self, all_res, *args, components='images', **kwargs):
+        """ Assemble the batch after a parallel action.
 
         Parameters
         ----------
-        components : str
-            a component name or names which data should be cropped
+        all_res : list
+            processed components of the batch
 
-        origin : tuple
-            can be one of:
+        components : str, sequence
+            names of the components to assemble
 
-            - tuple - a starting point in the form of (x, y)
-            - CROP_00 - to crop from left top edge (0,0)
-            - CROP_CENTER - to crop from center of each image
+        Returns
+        -------
+        self
+        """
+        _ = all_res, args, kwargs, components
+        raise NotImplementedError("Must be implemented in a child class")
 
-        shape : tuple
-            a crop size in the form of (width, height)
+    def _make_path(self, path, ix):
+        """ Compose path.
+
+        Parameters
+        ----------
+        path : str, None
+        ix : str
+            element's index (filename)
+
+        Returns
+        -------
+        path : str
+            joined path if path is not None else element's path specified in the batch's index
         """
-        if origin is not None or shape is not None:
-            origin = origin if origin is not None else (0, 0)
-            self._crop(components, origin, shape)
-        return self
+        return self.index.get_fullpath(ix) if path is None else os.path.join(path, ix)
 
-    @action
-    def random_crop(self, components='images', shape=None):
-        """ Crop all images to a given shape and a random origin
+    def _load_image(self, ix, src=None, components="images"):
+        """ Loads image
+
+        .. note:: only works with a single component
 
         Parameters
         ----------
+        path : str, None
+        ix : str
+            element's index (filename)
         components : str
-            a component name or names which data should be cropped
+            component to load
+
+        Returns
+        -------
+        np.ndarray : Loaded image.
+        """
+        _ = self, ix, src, components
+        raise NotImplementedError("Must be implemented in a child class")
+
+    @action
+    def load(self, *args, src=None, fmt=None, components=None, **kwargs):
+        """ Load data.
 
-        shape : tuple
-            a crop size in the form of (width, height)
+        .. note:: if `fmt='images'` than there must be single component.
```

Comment:
Where `there`?

Reply:
And leave a note that all parameters should be named only (no positional parameters allowed)

Reply:
+